From ed507cf84ee4cb462bdd19de67b188a02d9a2fe0 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Sun, 26 Jul 2026 00:11:44 +0800
Subject: [PATCH 17/52] UPSTREAM: elf: test: handle different rootsbindir in
 tst-ldconfig-cache

When compiling a glibc for a merged-/usr distro people may set
rootsbindir=/usr/sbin.  But tst-ldconfig-cache has hard-coded
/sbin/ldconfig path and so it fails with a different rootsbindir.

Fix it by using support_install_rootsbindir like run_ldconfig in
test-container.c.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
Reviewed-by: Florian Weimer <fweimer@redhat.com>
(cherry picked from commit 02ea17b5add83a205d8b204dce28f38fe0498ea3)
---
 elf/tst-ldconfig-cache.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/elf/tst-ldconfig-cache.c b/elf/tst-ldconfig-cache.c
index 9f71418b3a..f4820a1822 100644
--- a/elf/tst-ldconfig-cache.c
+++ b/elf/tst-ldconfig-cache.c
@@ -85,7 +85,10 @@ corrupt (void)
 static void
 ldconfig (void)
 {
-  xsystem ("/sbin/ldconfig -X");
+  char *cmd = xasprintf("%s/ldconfig -X", support_install_rootsbindir);
+  xsystem (cmd);
+
+  free(cmd);
 }
 
 /* Change ld.so.conf to refer to the new directory, and generate a new
-- 
2.55.0

